subsys/usb/usb_device.c

```diff
@@ -461,6 +461,45 @@ static bool usb_get_descriptor(u16_t type_index, u16_t lang_id,
 	return found;
 }
 
+static bool set_endpoint(const struct usb_ep_descriptor *ep_desc)
+{
+	struct usb_dc_ep_cfg_data ep_cfg;
+
+	ep_cfg.ep_addr = ep_desc->bEndpointAddress;
+	ep_cfg.ep_mps = sys_le16_to_cpu(ep_desc->wMaxPacketSize);
+
+	if (ep_desc->bmAttributes > USB_DC_EP_INTERRUPT) {
+		return false;
+	}
+
+	ep_cfg.ep_type = ep_desc->bmAttributes;
+
+	switch (ep_cfg.ep_type) {
+	case USB_DC_EP_BULK:
```

Comment:
```
if (defined(DT_USBHS_MAXIMUM_SPEED)) {
		if (ep_cfg.ep_mps > USB_MAX_BULK_HS_MPS) {
			return false;
		}
} else {
		if (ep_cfg.ep_mps > USB_MAX_BULK_FS_MPS) {
			return false;
		}
}
```

?

Reply:
I propose to use it like in the **header**:
```
#if defined(DT_USBHS_MAXIMUM_SPEED)
#define USB_MAX_BULK_MPS 512
#else
...
#endif
```

So we have this `DT_USBHS_MAXIMUM_SPEED` defined in DTS to use HS?

Reply:
BTW, for stm32 it is `DT_USB_MAXIMUM_SPEED`